Enhancements to MultilineAttributeDecorator

kecso opened the pull request:
- borderColor now handled as the background color of the whole box, while fillColor is used as the backgroundColor of individual textareas
- file drop is now handled and replaces the content of the textarea on drop (no blob involvement) *does not work on microsoft edge

kecso committed:
WIP - changes the multiline attribute decorator in two ways:
- borderColor now handled as the background color of the whole box, while fillColor is used as the backgroundColor of individual textareas
- file drop is now handled and replaces the content of the textarea on drop (no blob involvement) *does not work on microsoft edge

pmeijer commented on src/decorators/MultilineAttributeDecorator/DiagramDesigner/MultilineAttributeDecorator.DiagramDesignerWidget.js:
```diff
@@ -310,6 +310,21 @@ define([
                         self.client.setAttribute(self.nodeId, desc.name, $(this).val());
                     }
                 })
+                .on('drop', function (event) {
```
Isn't `'drop'` a more generic event than just file dropping?
What happens with the reader when the dropped file is a binary?

pmeijer replied:
Also it would be nice if the drop area reacted to a valid/invalid file drag over.

kecso committed:
#6 Fixing review comments.
- red/green background hints whether the drop will be accepted or not
- single file drop and text drop should work

pmeijer commented on src/decorators/MultilineAttributeDecorator/DiagramDesigner/MultilineAttributeDecorator.DiagramDesignerWidget.js:
```diff
@@ -23,7 +23,9 @@ define([
-        MIN_HEIGHT = 80;
+        MIN_HEIGHT = 80,
+        NO_DROP_COLOR = 'rgb(199,10,10)',
+        ALLOW_DROP_COLOR = 'rgb(10,199,10)';
```
Are these the same colors that are used through out the ui for drag and drop?

pmeijer commented on src/decorators/MultilineAttributeDecorator/DiagramDesigner/MultilineAttributeDecorator.DiagramDesignerWidget.js:
```diff
@@ -310,20 +312,32 @@ define([
+                    if (self._canAcceptTarget(event.originalEvent.dataTransfer)) {
+                        $(event.currentTarget).css('backgroundColor', ALLOW_DROP_COLOR);
+                    } else {
+                        $(event.currentTarget).css('backgroundColor', NO_DROP_COLOR);
```
Is `dragleave` triggered after a successful/unsuccessful drop? I don't see where the background is reset (unless the model is updated).

pmeijer commented on src/decorators/MultilineAttributeDecorator/DiagramDesigner/MultilineAttributeDecorator.DiagramDesignerWidget.js:
```diff
@@ -454,5 +468,33 @@ define([
+            reader;
+
+        if (dataTransfer.items[0].kind === 'string') {
+            callback(dataTransfer.getData('Text'));
```
:+1:

pmeijer reviewed (commented):
Have a couple of questions.

kecso committed:
#6 Fixing review comments.
Using same color schemes and handling color rendering in all cases.
